GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -669,10 +685,13 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
                         result = this.TryAddModifiedPath(gitUpdate.VirtualPath, isFolder: true);
                         if (result == FileSystemTaskResult.Success)
                         {
+                            this.newlyCreatedFileAndFolderPaths.Add(gitUpdate.VirtualPath);
+                            this.TryRemoveModifiedPath(gitUpdate.OldVirtualPath, isFolder: true);

```

Comment:
Should we check the return value of `TryRemoveModifiedPath` here?

Reply:
I was up in the air about that because on the one hand there won't be any issues if we fail to remove the path except the modified paths doesn't get as small.  On the other we could retry when needed and keep the modified paths small but I'm not sure we should kill the mount process if we get a `FatalError`.  What are you thoughts?

Reply:
I agree there's minimal harm to this failing, but I'm also thinking that if there are failures it could indicate some larger issue with the state of the file system (that I'm not sure we should ignore).

I lean toward checking the value (and killing the process if we see a `FatalError`).  @sanoursa, what are your thoughts?

Reply:
Yea I don't think we should ever ignore a `FatalError`. That said, why is `TryRemoveModifiedPath` ever returning `FatalError`s if we don't intend to shut down as a result? It feels like that method shouldn't even be a `Try` method - if we focus it down to tmp file scenarios, it can just be a best effort and not even return a result.

Reply:
It is still trying to write to the modified paths database file and the pattern is to retry if there was and IOException and return Fatal if there is some other exception.  To keep with the pattern I think we should shut down because we don't know what the state of the modified paths database file is in if we get a fatal error writing to it.